### PR TITLE
posix main: add left/right arrow to edit command

### PR DESCRIPTION
**Pull request by ChristophTobler**
with this, lines can be edited with the lef/right arrow key in a posix PX4 shell

### CODE REVIEW

File: src/platforms/posix/main.cpp [hunk 1]
```diff
@@ -502,9 +502,11 @@ int main(int argc, char **argv)
 
 	if (!daemon_mode) {
 		string mystr;
+		string add_string;
 		string string_buffer[CMD_BUFF_SIZE];
 		int buf_ptr_write = 0;
 		int buf_ptr_read = 0;
+		uint8_t cursor_position = 0;
```
Comment (bkueng) [inline src/platforms/posix/main.cpp]: Might be worth adding a note that it's counted from right to left (0 = end of string)

**Comment by ChristophTobler** [comment]
@bkueng Sure:+1: I'll add that as well

**Comment by ChristophTobler** [comment]
@bkueng Done.

**Comment by bkueng** [comment]
There are multiple keycodes (https://github.com/mintty/mintty/wiki/Keycodes#editing-keys), I added them as well.

**Comment by mhkabir** [comment]
<3

**Comment by dagar** [comment]
In the past few days I've been seeing this in ROS mission test output.

image


http://ci.px4.io:8080/blue/organizations/jenkins/Firmware/detail/coverity_scan/59/pipeline/

Could it be related to the shell changes?

**Comment by bkueng** [comment]
@dagar https://github.com/PX4/Firmware/pull/8704 should fix this.

**Pull request merged**